platform/ui/src/contextProviders/DialogProvider.jsx

```diff
@@ -302,3 +314,27 @@ DialogProvider.propTypes = {
 };
 
 export default DialogProvider;
+
+function OutsideAlerter(props) {
+  const wrapperRef = useRef(null);
+
+  useEffect(() => {
+    /**
+     * Alert if clicked on outside of element
+     */
+    function handleClickOutside(event) {
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+        props.onClickOutside();
+      }
+    }
+
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
```

Comment:
what about touch devices? We should probably add touchstart as well

Reply:
Can you close dialogs via Esc?

Reply:
Good shout on touchstart, yeah esc works on the dialogs.